### PR TITLE
initial content for 2.6.7

### LINKED ISSUE #6430: Release task: What's new in 2.6.7 (with 2.6.6)
Similar to https://github.com/rancher/dashboard/issues/5638, this ticket is to update dashboard/content/docs/en-us/whats-new.md with material that appears in the UI about what's new. This was missed in 2.6.6.

Checking with @wjimenez5271 what needs to go in.

(Normally there would be a ticket like https://github.com/rancher/rancher/issues/37315 for the overall release, so I'm jumping the gun a bit.)

### PR DESCRIPTION
**Pull request by gaktive**
Signed-off-by: Gary A Korhonen <gkorhonen@suse.com>

Fixes https://github.com/rancher/dashboard/issues/6430
Includes 2.6.6 blurb though there was only one bug addressed

### Summary
Fixes #6430 

Short summary of changes in Rancher 2.6.7 (plus including potential copy for 2.6.6) to be shown in the UI

### Areas or cases that should be tested

Verify that What's New shows the expected copy

### CODE REVIEW

**Comment by gaktive** [comment]
This is presently a draft. Stand by for future updates and proofreading.

**Comment by gaktive** [comment]
@deniseschannon the rest is up to you for both 2.6.7 and if we want anything for 2.6.6

File: shell/content/docs/en-us/whats-new.md
Comment (snasovich) [inline shell/content/docs/en-us/whats-new.md:22]:
```suggestion
- RKE2 encrypts secret data at rest by default. You can now rotate the key used for this encryption for Rancher-provisioned RKE2 clusters with supported k8s versions.
```

@jakefhyde @Oats87 , please correct / add as necessary

File: shell/content/docs/en-us/whats-new.md
Comment (snasovich) [inline shell/content/docs/en-us/whats-new.md:10]: Nit, but I think we want to match capitalization of official feature name
```suggestion
- Now you can purchase support for Rancher through the AWS Marketplace 
```

File: shell/content/docs/en-us/whats-new.md
Comment (snasovich) [inline shell/content/docs/en-us/whats-new.md:18]: We support running Rancher itself on 1.24. We also support it for downstream provisioned RKE2 clusters (not just imported ones).

So, I think we should state something like this:
```suggestion
- Kubernetes 1.24 is now supported Kubernetes version for running Rancher as well for provisioned and imported RKE1/RKE2/k3s clusters.
```

@jiaqiluo @kinarashah @sowmyav27 @slickwarren, please chime in if the above is incorrect.

Comment (jiaqiluo) [reply shell/content/docs/en-us/whats-new.md:18]: The proposed changes LGTM.  
A side note is that now 1.24 is the default version when creating an RKE1/RKE2/K3s cluster via Rancher UI and the default version for RKE CLI.

File: shell/content/docs/en-us/whats-new.md
Comment (jakefhyde) [reply shell/content/docs/en-us/whats-new.md:22]: It might be worth calling out that this feature is also available for k3s as well (although encryption not enabled by default) for clarity. Also I know we are release noting https://github.com/rancher/rke2/issues/3186, but should we also call it out here as well?

Comment (snasovich) [reply shell/content/docs/en-us/whats-new.md:22]: @jakefhyde , thank you!

I believe we generally want short statements on UI splash screen highlighting the features. So I don't believe we want to list all known issues and such here. But I'll let @gaktive decide.
Also, the actual issue that we will be release noting is this one: https://github.com/rancher/rancher/issues/38283

File: shell/content/docs/en-us/whats-new.md
Comment (slickwarren) [reply shell/content/docs/en-us/whats-new.md:18]: I agree, LGTM

File: shell/content/docs/en-us/whats-new.md
Comment (cbron) [reply shell/content/docs/en-us/whats-new.md:10]:
```suggestion
- You can now purchase support for Rancher directly through the AWS Marketplace:  https://rancher.com/docs/rancher/v2.6/en/installation/cloud-marketplace/aws/
```

That link will be dead until docs are merged, you can see it here: https://staging.rancher.com/docs/rancher/v2.6/en/installation/cloud-marketplace/aws/

Maybe @wjimenez5271 has the link to the actual csp-adapter listing.

File: shell/content/docs/en-us/whats-new.md
Comment (cbron) [inline shell/content/docs/en-us/whats-new.md:29]: Is this for 4484 ? That was released in 2.6.6, but obviously this modal wasn't updates, so is this aiming to cover that ?

File: shell/content/docs/en-us/whats-new.md
Comment (cbron) [inline shell/content/docs/en-us/whats-new.md:14]:
```
Microsoft have deprecated the Azure AD Graph API that Rancher had been using for authentication via Azure AD. 
A configuration update is necessary to make sure users can still use Rancher with Azure AD.
For more details see: https://rancher.com/docs/rancher/v2.6/en/admin-settings/authentication/azure-ad/#migrating-from-azure-ad-graph-api-to-microsoft-graph-api
```

Taken and modified from: https://github.com/rancher/rancher/issues/29306#issuecomment-1195846938

File: shell/content/docs/en-us/whats-new.md
Comment (gaktive) [reply shell/content/docs/en-us/whats-new.md:29]: Yeah, I want to primarily account for why there's a 2.6.6 in this area since we'd skip from 2.6.5 to 2.6.7 without any explanation.

File: shell/content/docs/en-us/whats-new.md
Comment (gaktive) [reply shell/content/docs/en-us/whats-new.md]: Thanks!

File: shell/content/docs/en-us/whats-new.md
Comment (cbron) [inline shell/content/docs/en-us/whats-new.md:10]: We should still add something here.

File: shell/content/docs/en-us/whats-new.md
Comment (cbron) [inline shell/content/docs/en-us/whats-new.md:29]: You'll need to adjust title/section above to all make sense together: 

```2.6.6 was released to account for a major performance issue. The issue occurred when Rancher was attempting to control large volumes of traffic from downstream clusters. This mechanism was not handling disconnects properly and would result in indefinite locks. This is now working properly.```

Comment (gaktive) [reply shell/content/docs/en-us/whats-new.md]: Thanks!

File: shell/content/docs/en-us/whats-new.md
Comment (gaktive) [reply shell/content/docs/en-us/whats-new.md]: Done in another conversation.

File: shell/content/docs/en-us/whats-new.md
Comment (cbron) [reply shell/content/docs/en-us/whats-new.md]: Actually we shouldn't do that since there are different listings for diff countries. I like text above. Reposting: 

```
You can now purchase support for Rancher directly through the AWS Marketplace:  
https://rancher.com/docs/rancher/v2.6/en/installation/cloud-marketplace/aws/
```

File: shell/content/docs/en-us/whats-new.md
Comment (gaktive) [reply shell/content/docs/en-us/whats-new.md]: Per an earlier note, we'll go for:
> You can now purchase support for Rancher directly through the AWS Marketplace:  
https://rancher.com/docs/rancher/v2.6/en/installation/cloud-marketplace/aws/

**Review by richard-cox — COMMENTED** [review]
If you rebase it'll remove the failing FOSSA check (we need to go back to EIO and address this when 2.6.7 is out), otherwise LGTM

**Review by richard-cox — APPROVED** [review]

**Pull request merged**
